File: Permutation-Difference-between-Two-Strings.cpp
class Solution {
public:
    int findPermutationDifference(string s, string t) {
        int p[26];
        for(int i=0;i<s.size();i++){
            p[s[i]-'a']=i;
        }
        int sumd=0;
        for(int i=0;i<t.size();i++){
            sumd += abs(p[t[i]-'a']-i);
        }
        return sumd;
    }
};

File: Permutation-Difference-between-Two-Strings.java
class Solution {
    public int findPermutationDifference(String s, String t) {
        int[] p = new int[26]; 
        for (int i = 0; i < s.length(); i++) {
            p[s.charAt(i) - 'a'] = i;
        }
        int sumd = 0; 
        for (int i = 0; i < t.length(); i++) {
            sumd += Math.abs(p[t.charAt(i) - 'a'] - i); // Calculate and sum differences
        }
        return sumd; // Return the total sum of differences
    }
}